src/main/java/group1/mips_simulator/components/cpuParts/Register.java

```diff
@@ -9,13 +9,30 @@
 public class Register {
     public Value value = new Value(0);
 
-    public Value read() {
+    //getter methods
+    public Value get() {
+        // get the Value in the register
         return this.value;
     }
+    public short read() {
+        // read the number in the register
+        return this.value.get();
+    }
+
 
-    public void write(Value newValue) {
+    // setter methods
+    public void set(Value newValue) {
+        // set the Value of the register
         this.value = newValue;
     }
+    public void write(short value_) {
+        // write a short to the register
+        this.value = new Value(value_);
```

Comment:
Typically, a more elegant way to implement these kinds of functions would be to only implement it once, say the `set(Value newValue)` one is the only one that actually has the `this.value = newValue;` line.

Then each of these other ones will simply call the one actual functional implementation. So the `write(short value_) { set(new Value(value)); }`

This way, if we ever want to make changes or updates we only need to update it in one place instead of 3. So for example, suppose we wanted to add validation to make sure the `newValue` item can fit into the register (ie it must be in the range [-32,768 to 32,767]). Then we'd only need to add that check to the one actual implemented function and all the other functions will "dispatch" to it.

Reply:
I getchu, so that theres only one method that actually sets, and every other method just encapsulates it right?
I can modify this to that, good call

Reply:
Yep! It's a minor design pattern detail, not critical, but I thought I'd point it out now.